addons/website_livechat/models/mail_channel.py

```diff
@@ -28,6 +28,8 @@ def channel_info(self, extra_info=False):
         at the top of the livechat channel discussion view in discuss module.
         """
         channel_infos = super(MailChannel, self).channel_info(extra_info)
+        if not self.env['website.visitor'].check_access_rights('read', raise_exception=False):
```

Comment:
Also if you keep the "check first" version, you would need to also call `check_access_rule` (on principle at least, I'm not sure whether visitor actually has specific rules that would make this an issue).